src/browserbehavior/DefaultBrowserBehavior.ts

```diff
@@ -117,6 +117,10 @@ export default class DefaultBrowserBehavior implements BrowserBehavior {
     return this.isSafari();
   }
 
+  requiresNoExactMediaStreamConstraints(): boolean {
+    return this.isIOSSafari() && (this.version() === '12.0.0' || this.version() === '12.1.0');
```

Comment:
Is this only for iOS Safari or can it be for desktop Safari?
Also is this for any 12.0+ and 12.1+ or just 12.0.0 and 12.1.0?

Reply:
Yeah, I tested with macOS safari 12.0 and it has a exact for deviceId.